src/components/component-placeholder/component-placeholder.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+interface ComponentPlaceholderProps {
+    name: string;
+    height?: string;
+}
+
+interface ComponentPlaceholderState {
+    componentNumber: number;
+}
+
+class ComponentPlaceholder extends React.Component<ComponentPlaceholderProps, ComponentPlaceholderState> {
+
+    static componentCount: number = 0;
+
+    static bgColors = [
+        '#878',
+        '#788',
+        '#887',
+        '#9b9',
+        '#e99',
+        '#99b'
+    ];
+
+    constructor(props: ComponentPlaceholderProps, state: ComponentPlaceholderState) {
+        super(props, state);
+        this.state = {
+            componentNumber: ComponentPlaceholder.componentCount
+        };
+        ComponentPlaceholder.componentCount++;
+    }
+
+    public render() {
+        const bgColor =
+            ComponentPlaceholder.bgColors[this.state.componentNumber % ComponentPlaceholder.bgColors.length];
+        const stripeColor = '#eee';
+
+        const PlaceholderDiv = styled.div`
+        background:
+         repeating-linear-gradient(45deg, ${bgColor}, ${bgColor} 1.8em, ${stripeColor} 1.8em, ${stripeColor} 2em);
+        height: ${this.props.height === undefined ? '100%' : this.props.height};
+        display: flex;
+        align-items: center;
+        justify-content: center;
+    `;
+
+        const VerticalTitle = styled.h1`
```

Comment:
Vil tro at parsing og oppretting av komponenter sånn er en relativt tung for CPUen, så der det er mulig bør nok disse flyttes ut av `render`.

Reply:
ser den. vet ikke hva som er vanlig her, men en av fordelene med styled-components er at man kan bruke props for å sette css-atributter, og da må de lages et sted de har tilgang på props. Kan forske litt på hvordan dette kan løses på en god måte.